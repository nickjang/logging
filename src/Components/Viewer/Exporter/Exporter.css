.exporter {
  background-color: lightgray;
  margin: 17px 0px;
}

.exporter #export-heading + fieldset {
  height: fit-content;
}