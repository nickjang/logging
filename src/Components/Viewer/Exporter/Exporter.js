import React, { Component } from 'react';
import ExportOption from '../ExportOption/ExportOption';

class Exporter extends Component {
  state = {
    orientation: '',
    selectedOptions: [],
    exporting: false,
    fetchError: {
      code: '',
      message: ''
    }
  }

  orientationRef = React.createRef();

  updateOrientation = (orientation) => {
    this.setState({ orientation });
  }

  updateSelectedOptions = (option) => {
    this.setState({ 
      selectedOptions: [...this.state.selectedOptions, option] 
    });
  }

  handleSubmit = (e) => {
    e.preventDefault();
    //exporting/error
    //this.props.exportLogs(orientation, selectedOptions)
  }

  /**
   * Render radio button for row and column
   * options.
   */
  renderOrientationOptions = () => {
    return ['As columns', 'As rows']
      .map(option => {
        return (
          <ExportOption
            key={option}
            type='radio'
            ref={this.orientationRef}
            option={option}
            name='orientation'
            form='export-form'
            selected={this.state.orientation}
            update={this.updateOrientation}
            role='presentation' />
        );
      });
  }

  /**
   * Render checkbox for each export format.
   */
  renderExportOptions = () => {
    const exportOptions = [
      'Excel', 'SQL', 'Dataframe',
      'TXT', 'Array', 'Formatted String',
      'Object', 'CSV', 'JSON'
    ].map(option => {
      return (
        <ExportOption
          key={option}
          type='checkbox'
          option={option}
          name='export-options'
          form='export-form'
          selected={this.state.selectedOptions}
          update={this.updateSelectedOptions}
          role='gridcell' />
      );
    });

    // Position checkboxes into rows
    let rows = [];
    let perRow = 3;
    for (let i = 0; i < exportOptions.length; i += perRow) {
      rows.push(
        <div class='group-row' role='row'>
          exportOptions.slice(i, i + perRow)
        </div>
      );
    }
    return rows;
  }

  render() {
    return (
      <section class='export-logs'>
        <output form='export-form' className='form-status'>{this.state.fetchError.message || (this.state.exporting && 'Exporting...')}</output>
        <div class='group-row' role='presentation'> {/*<!--Check how this affects DOM-->*/}
          <h3 id='export-heading'>Export Logs</h3>
          <fieldset class='group-column'>
            {this.renderOrientationOptions()}
          </fieldset>
        </div>
        <form action='' id='export-form' class='group-column' role='grid' aria-labelledby='export-heading'>
          {this.renderExportOptions()}
          <button
            type='submit'
            form='export-form'
            onClick={(e) => { this.handleSubmit(e) }}
            disabled={this.state.orientation || this.state.selectedOptions.length}
          > Export
          </button>
        </form>
      </section>
    );
  }
}

export default Exporter;