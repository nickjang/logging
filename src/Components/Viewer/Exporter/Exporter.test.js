import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Exporter from './Exporter';

describe(`Exporter component`, () => {
  it('renders Exporter by default', () => {
    const wrapper = shallow(<Exporter />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})