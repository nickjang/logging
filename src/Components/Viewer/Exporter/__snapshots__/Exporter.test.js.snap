// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exporter component renders Exporter by default 1`] = `
<section
  className="exporter lg-card"
>
  <output
    className="form-status"
    form="export-form"
  />
  <div
    className="group-row"
    role="presentation"
  >
     
    <h3
      className="lg-title"
      id="export-heading"
    >
      Export Logs
    </h3>
    <fieldset
      className="group-column m-1"
    >
      <ExportOption
        form="export-form"
        inputRef={null}
        key="As columns"
        name="orientation"
        option="As columns"
        role="presentation"
        selected={
          Array [
            "",
          ]
        }
        type="radio"
        update={[Function]}
      />
      <ExportOption
        form="export-form"
        inputRef={null}
        key="As rows"
        name="orientation"
        option="As rows"
        role="presentation"
        selected={
          Array [
            "",
          ]
        }
        type="radio"
        update={[Function]}
      />
    </fieldset>
  </div>
  <form
    action=""
    aria-labelledby="export-heading"
    className="group-column"
    id="export-form"
    role="grid"
  >
    <fieldset
      className="group-row"
      form="export-form"
    >
      <div
        key="export-options-col-0"
        role="row"
      >
        <ExportOption
          form="export-form"
          inputRef={null}
          key="Excel"
          name="export-options"
          option="Excel"
          role="gridcell"
          type="checkbox"
          update={[Function]}
        />
        <ExportOption
          form="export-form"
          inputRef={null}
          key="SQL"
          name="export-options"
          option="SQL"
          role="gridcell"
          type="checkbox"
          update={[Function]}
        />
        <ExportOption
          form="export-form"
          inputRef={null}
          key="Dataframe"
          name="export-options"
          option="Dataframe"
          role="gridcell"
          type="checkbox"
          update={[Function]}
        />
      </div>
      <div
        key="export-options-col-3"
        role="row"
      >
        <ExportOption
          form="export-form"
          inputRef={null}
          key="TXT"
          name="export-options"
          option="TXT"
          role="gridcell"
          type="checkbox"
          update={[Function]}
        />
        <ExportOption
          form="export-form"
          inputRef={null}
          key="Array"
          name="export-options"
          option="Array"
          role="gridcell"
          type="checkbox"
          update={[Function]}
        />
        <ExportOption
          form="export-form"
          inputRef={null}
          key="Formatted String"
          name="export-options"
          option="Formatted String"
          role="gridcell"
          type="checkbox"
          update={[Function]}
        />
      </div>
      <div
        key="export-options-col-6"
        role="row"
      >
        <ExportOption
          form="export-form"
          inputRef={null}
          key="Object"
          name="export-options"
          option="Object"
          role="gridcell"
          type="checkbox"
          update={[Function]}
        />
        <ExportOption
          form="export-form"
          inputRef={null}
          key="CSV"
          name="export-options"
          option="CSV"
          role="gridcell"
          type="checkbox"
          update={[Function]}
        />
        <ExportOption
          form="export-form"
          inputRef={null}
          key="JSON"
          name="export-options"
          option="JSON"
          role="gridcell"
          type="checkbox"
          update={[Function]}
        />
      </div>
    </fieldset>
    <div
      className="lg-text-right mt-1"
    >
      <button
        className="lg-btn lg-btn-success"
        disabled={true}
        form="export-form"
        onClick={[Function]}
        type="submit"
      >
         Export
      </button>
    </div>
  </form>
</section>
`;
